src/Mod/RuleSoldier.cpp

```diff
@@ -178,16 +178,50 @@ void RuleSoldier::load(const YAML::Node &node, Mod *mod, int listOrder, const Mo
 	mod->loadSpriteOffset(_type, _rankSprite, node["rankSprite"], "BASEBITS.PCK");
 	mod->loadSpriteOffset(_type, _rankSpriteBattlescape, node["rankBattleSprite"], "SMOKE.PCK");
 	mod->loadSpriteOffset(_type, _rankSpriteTiny, node["rankTinySprite"], "TinyRanks");
-
+	mod->loadSpriteOffset(_type, _skillIconSprite, node["skillIconSprite"], "SPICONS.DAT");
+	
 	_listOrder = node["listOrder"].as<int>(_listOrder);
 	if (!_listOrder)
 	{
 		_listOrder = listOrder;
 	}
+	
+	auto loadSkillActionConf = [&](int offset, const YAML::Node &n)
+	{
+		if (n)
+		{
+			RuleSkill *skill = new RuleSkill(offset);
+			
+			loadCost(skill->Cost, n, "Use");
+			loadPercent(skill->Flat, n, "Use");
+			
+			skill->Name = n["name"].as<std::string>(skill->Name);
+			skill->RequiredBonus = node["requiredBonus"].as<std::vector<std::string>>(skill->RequiredBonus);
+			int targetMode = n["targetMode"].as<int>(skill->TargetMode);
+			targetMode = targetMode < 0 ? 0 : targetMode;
+			targetMode = targetMode > BA_CQB ? 0 : targetMode;
+			skill->TargetMode = static_cast<BattleActionType>(targetMode);
+			skill->IsPsiRequired = n["isPsiRequired"].as<bool>(skill->IsPsiRequired);
+			skill->CheckHandsOnly = n["checkHandsOnly"].as<bool>(skill->CheckHandsOnly);
+			skill->CompatibleWeapons = n["compatibleWeapons"].as<std::vector<std::string>>(skill->CompatibleWeapons);
```

Comment:
btw this code leak, overall I think skill should be separate rule object that is fully exposed to scripts like armors.

and soldiers only have reference to it like:
```
soldier:
  - name: X
     skills:
    - A
    - B
    - C
```
As bonuses we could add it to armors too.

and after all mod are loaded you link directly soldier with correct skills.

Reply:
> Do we really need this limitation? This make hard to mix mods, because only one list can be loaded.
> I think better would be to allow all "compatible" weapons to be use, this mean have selected battle type and attack type.
> To make more limitations, script can be used to check if that item is compatible. It would be more extendable as many mods could add same tags, or even add new script that will override it.

Hmm, I wanted to allow as much of the functioniality to be usable without scripts as possible.
And I definitely wanted to have a method of selecting items which is much more precise than battle type/action. Also the action is currently definable per skill, allowing override of item actions, which allows for interesting effects which have to be tailor made for a specific item.

Reply:
> btw this code leak, overall I think skill should be separate rule object that is fully exposed to scripts like armors.

I am no C++ coder, so please forgive my missing knowledge in memory management. Please explain to me where this code leaks, so I can learn from it.

> and soldiers only have reference to it like:
> 
> ```
> soldier:
>   - name: X
>      skills:
>     - A
>     - B
>     - C
> ```
> 
> As bonuses we could add it to armors too.
> 
> and after all mod are loaded you link directly soldier with correct skills.

This sounds like an interesting idea and could make it potentially easier to allow the AI to use skills someday.

Reply:
> 
> 
> > Do we really need this limitation? This make hard to mix mods, because only one list can be loaded.
> > I think better would be to allow all "compatible" weapons to be use, this mean have selected battle type and attack type.
> > To make more limitations, script can be used to check if that item is compatible. It would be more extendable as many mods could add same tags, or even add new script that will override it.
> 
> Hmm, I wanted to allow as much of the functioniality to be usable without scripts as possible.
> And I definitely wanted to have a method of selecting items which is much more precise than battle type/action. Also the action is currently definable per skill, allowing override of item actions, which allows for interesting effects which have to be tailor made for a specific item. What about allowing both? A selection by weapon name (type) and a selection by battle_type?

My problem is that current versions is hard to scale. Image 6 mods that try add skills and weapons. And how they could be independent to each other? In first draft it was near impossible to to do something like that. Best case would be if you could enable each mode separately and each mod can work on its own, but if you enable them at once they all will integrate smoothly. This was reason why I implemented Tags that can be declared in every mode and game check if each definitions is same.

And for memory, simply you used `new` but `delete` was never used. This mean memory leak, even for Rule objects that live for whole program life (with exception of mod reload).